src/neo/SmartContract/Native/RoleManagement.cs

```diff
@@ -59,6 +59,7 @@ private void DesignateAsRole(ApplicationEngine engine, Role role, ECPoint[] node
             list.AddRange(nodes);
             list.Sort();
             engine.Snapshot.Add(key, new StorageItem(list));
+            engine.SendNotification(Hash, "designation", new VM.Types.Array(engine.ReferenceCounter, new StackItem[] { (int)role }));
```

Comment:
How about adding block `index` into notification?

Reply:
I think that it's and information that it's not added in other events, it's easy to get, what do you think @erikzhang ?

Reply:
Block number really is important for this (to issue proper `getDesignatedByRole` call using notification data only) . And keys would also be convenient to have.

Reply:
It is always the same as the index of the block where the notification is located, right? If so I don't think we need it.

Reply:
You may be receiving just notification data without any block/transaction metadata, you won't get block index in this case unless it's included in the notification.